diff --git a/boot/espressif/hal/CMakeLists.txt b/boot/espressif/hal/CMakeLists.txt
index d248c267..eb743e12 100644
--- a/boot/espressif/hal/CMakeLists.txt
+++ b/boot/espressif/hal/CMakeLists.txt
@@ -14,6 +14,7 @@ set(include_dirs
     )
 
 list(APPEND include_dirs
+    ${esp_hal_dir}/components/esp_bootloader_format/include
     ${esp_hal_dir}/components/bootloader_support/include
     ${esp_hal_dir}/components/bootloader_support/private_include
     ${esp_hal_dir}/components/bootloader_support/bootloader_flash/include
@@ -38,7 +39,7 @@ list(APPEND include_dirs
     ${esp_hal_dir}/components/esp_hw_support/include/soc/${MCUBOOT_TARGET}
     ${esp_hal_dir}/components/esp_hw_support/port/include
     ${esp_hal_dir}/components/esp_hw_support/include/esp_private
-    ${esp_hal_dir}/components/esp_hw_support/port/${MCUBOOT_TARGET}
+    ${esp_hal_dir}/components/esp_hw_support/port/${MCUBOOT_TARGET}/include
     ${esp_hal_dir}/components/hal/${MCUBOOT_TARGET}/include
     ${esp_hal_dir}/components/hal/include
     ${esp_hal_dir}/components/hal/platform_port/include
@@ -73,7 +74,7 @@ set(hal_srcs
     ${esp_hal_dir}/components/bootloader_support/src/esp_image_format.c
     ${esp_hal_dir}/components/bootloader_support/src/${MCUBOOT_TARGET}/bootloader_soc.c
     ${esp_hal_dir}/components/bootloader_support/src/${MCUBOOT_TARGET}/bootloader_sha.c
-    ${esp_hal_dir}/components/hal/mpu_hal.c
+#   ${esp_hal_dir}/components/hal/mpu_hal.c
     ${esp_hal_dir}/components/hal/efuse_hal.c
     ${esp_hal_dir}/components/hal/mmu_hal.c
     ${esp_hal_dir}/components/hal/wdt_hal_iram.c
diff --git a/boot/espressif/hal/include/esp_log.h b/boot/espressif/hal/include/esp_log.h
index ad6270d7..5cde0621 100644
--- a/boot/espressif/hal/include/esp_log.h
+++ b/boot/espressif/hal/include/esp_log.h
@@ -26,4 +26,6 @@
 #define ESP_EARLY_LOGD(tag, fmt, ...) MCUBOOT_LOG_DBG("[%s] " fmt, tag, ##__VA_ARGS__)
 #define ESP_EARLY_LOGV(tag, fmt, ...) MCUBOOT_LOG_DBG("[%s] " fmt, tag, ##__VA_ARGS__)
 
+#define ESP_LOG_INFO MCUBOOT_LOG_LEVEL_INFO
+
 uint32_t esp_log_early_timestamp(void);
diff --git a/boot/espressif/port/esp32c3/bootloader.conf b/boot/espressif/port/esp32c3/bootloader.conf
index 88954eea..8564cb44 100644
--- a/boot/espressif/port/esp32c3/bootloader.conf
+++ b/boot/espressif/port/esp32c3/bootloader.conf
@@ -6,10 +6,10 @@ CONFIG_ESP_FLASH_SIZE=4MB
 CONFIG_ESP_BOOTLOADER_SIZE=0xF000
 CONFIG_ESP_BOOTLOADER_OFFSET=0x0000
 CONFIG_ESP_IMAGE0_PRIMARY_START_ADDRESS=0x10000
-CONFIG_ESP_APPLICATION_SIZE=0x100000
-CONFIG_ESP_IMAGE0_SECONDARY_START_ADDRESS=0x110000
+CONFIG_ESP_APPLICATION_SIZE=0x150000
+CONFIG_ESP_IMAGE0_SECONDARY_START_ADDRESS=0x160000
 CONFIG_ESP_MCUBOOT_WDT_ENABLE=y
-CONFIG_ESP_SCRATCH_OFFSET=0x210000
+CONFIG_ESP_SCRATCH_OFFSET=0x2B0000
 CONFIG_ESP_SCRATCH_SIZE=0x40000
 
 # When enabled, prevents updating image to an older version
@@ -43,15 +43,15 @@ CONFIG_ESP_SCRATCH_SIZE=0x40000
 # CONFIG_ESP_SERIAL_BOOT_GPIO_TX=9
 
 # Use UART0 for console printing (use either UART or USB alone)
-CONFIG_ESP_CONSOLE_UART=y
-CONFIG_ESP_CONSOLE_UART_NUM=0
+# CONFIG_ESP_CONSOLE_UART=y
+# CONFIG_ESP_CONSOLE_UART_NUM=0
 # Configures alternative UART port for console printing
 # (UART_NUM=0 must not be changed)
 # CONFIG_ESP_CONSOLE_UART_CUSTOM=y
 # CONFIG_ESP_CONSOLE_UART_TX_GPIO=9
 # CONFIG_ESP_CONSOLE_UART_RX_GPIO=8
 # Use USB JTAG Serial for console printing
-# CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG=y
+CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG=y
 
 # CONFIG_ESP_SIGN_EC256=y
 # CONFIG_ESP_SIGN_ED25519=n
